### PR TITLE
Add SpectrumInfo and twoTheta python exports

### LINKED ISSUE #21817: New python exports needed for SpectrumInfo
mslice has a need for the `twotheta` method of `SpectrumInfo` to be exported to python, so the scattering angle information can be accessed from an `MDEventWorkspace.`

### PR DESCRIPTION
**Pull request by MatthewBowles**
[Mslice](https://github.com/mantidproject/mslice) recently required access to scattering angle information through `SpectrumInfo` (due to using a non-matrix workspace). This PR adds the required exports.

**To test:**
Test `SpectrumInfo` and `twoTheta` can be used in python e.g. `workspace.spectrumInfo().twoTheta(x)`

Fixes #21817

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)?
- [ ] Are the unit tests small and test the class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] Do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by quantumsteve** [comment]
Have you considered discussions in https://github.com/mantidproject/mantid/pull/18962 and https://github.com/mantidproject/mantid/pull/19019?

**Comment by MatthewBowles** [comment]
Yes. `getDetector` cannot be used in mslice's use case since I am not operating on a `MatrixWorkspace`. `SpectrumInfo` seems to be in a more stable state now, so I see no reason not to add exports (since it was planned to do this eventually anyway). I have discussed with @SimonHeybrock about this.

File: Framework/PythonInterface/mantid/api/src/Exports/ExperimentInfo.cpp [hunk 1]
```diff
@@ -81,5 +82,9 @@ void export_ExperimentInfo() {
       .def("detectorInfo", &ExperimentInfo::detectorInfo,
            return_value_policy<reference_existing_object>(), args("self"),
            "Return a const reference to the :class:`~mantid.api.DetectorInfo` "
+           "object.")
+      .def("spectrumInfo", &ExperimentInfo::spectrumInfo,
```
Comment (SimonHeybrock) [inline Framework/PythonInterface/mantid/api/src/Exports/ExperimentInfo.cpp]: I think this is currently in the child class export `MatrixWorkspace`, so if we add it here in the base, remove it in the child.

**Review by SimonHeybrock — CHANGES_REQUESTED** [review]
One comment:

File: Framework/PythonInterface/mantid/api/src/Exports/MatrixWorkspace.cpp [hunk 2]
```diff
@@ -243,11 +243,6 @@ void export_MatrixWorkspace() {
       .def("YUnitLabel", &MatrixWorkspace::YUnitLabel, arg("self"),
            "Returns the caption for the Y axis")
 
-      .def("spectrumInfo", &MatrixWorkspace::spectrumInfo,
```
Comment (SimonHeybrock) [inline Framework/PythonInterface/mantid/api/src/Exports/MatrixWorkspace.cpp]: Also remove the include.

**Review by SimonHeybrock — APPROVED** [review]

**Pull request merged**